engine/src/main/java/org/hibernate/search/engine/search/projection/spi/ProjectionTypeKeys.java

```diff
@@ -20,5 +20,10 @@ public static <T> SearchQueryElementTypeKey<T> key(String name) {
 	public static final SearchQueryElementTypeKey<FieldProjectionBuilder.TypeSelector> FIELD = key( "field" );
 	public static final SearchQueryElementTypeKey<DistanceToFieldProjectionBuilder> DISTANCE = key( "distance" );
 	public static final SearchQueryElementTypeKey<CompositeProjectionBuilder> OBJECT = key( "object" );
-
+	public static final SearchQueryElementTypeKey<CompositeProjectionBuilder> ID = key( "id" );
+	public static final SearchQueryElementTypeKey<CompositeProjectionBuilder> DOCUMENT_REFERENCE = key( "document reference" );
+	public static final SearchQueryElementTypeKey<CompositeProjectionBuilder> ENTITY = key( "entity" );
+	public static final SearchQueryElementTypeKey<CompositeProjectionBuilder> ENTITY_REFERENCE = key( "entity reference" );
```

Comment:
I'd avoid spaces in keys, and since you don't use the builder you might as well hide its type.

```suggestion
	public static final SearchQueryElementTypeKey<?> ID = key( "id" );
	public static final SearchQueryElementTypeKey<?> DOCUMENT_REFERENCE = key( "document-reference" );
	public static final SearchQueryElementTypeKey<?> ENTITY = key( "entity" );
	public static final SearchQueryElementTypeKey<?> ENTITY_REFERENCE = key( "entity-reference" );
```